Apps/phcrowdstrikeoauth/crowdstrikeoauthapi_connector.py

```diff
@@ -1002,26 +1015,86 @@ def _handle_list_groups(self, param):
     def _handle_list_custom_indicators(self, param):
 
         self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        # Add an action result to the App Run
         action_result = self.add_action_result(ActionResult(dict(param)))
 
-        params = {}
+        api_data = {
+            "limit": 400  # 500 is the max, don't want to give max, this could be tuned
+        }
 
-        if param.get('indicator_value'):
-            params['values'] = param['indicator_value']
-        if param.get('indicator_type'):
-            params['types'] = param['indicator_type']
+        # optional parameters
+        if CROWDSTRIKE_JSON_LIST_IOC in param:
+            api_data["values"] = [param[CROWDSTRIKE_JSON_LIST_IOC]]
+        if CROWDSTRIKE_IOCS_POLICY in param and param[CROWDSTRIKE_IOCS_POLICY] != "all":
+            api_data["policies"] = [param[CROWDSTRIKE_IOCS_POLICY]]
+        if CROWDSTRIKE_IOCS_SHARE_LEVEL in param and param[CROWDSTRIKE_IOCS_SHARE_LEVEL] != "all":
+            api_data["share_levels"] = param[CROWDSTRIKE_IOCS_SHARE_LEVEL]
+        if CROWDSTRIKE_SEARCH_IOCS_FROM_EXPIRATION in param:
+            api_data["from.expiration_timestamp"] = param[CROWDSTRIKE_SEARCH_IOCS_FROM_EXPIRATION]
+        if CROWDSTRIKE_SEARCH_IOCS_TO_EXPIRATION in param:
+            api_data["to.expiration_timestamp"] = param[CROWDSTRIKE_SEARCH_IOCS_TO_EXPIRATION]
+        if CROWDSTRIKE_IOCS_SOURCE in param:
+            api_data["sources"] = param[CROWDSTRIKE_IOCS_SOURCE]
+        if CROWDSTRIKE_SEARCH_IOCS_TYPE in param and param[CROWDSTRIKE_SEARCH_IOCS_TYPE] != "all":
+            api_data["types"] = param[CROWDSTRIKE_SEARCH_IOCS_TYPE]
+            if param[CROWDSTRIKE_SEARCH_IOCS_TYPE] == "hash":
+                api_data["types"] = ["md5", "sha1", "sha256"]
+
+        more = True
+
+        self.send_progress("Completed 0 %")
+        data = defaultdict(list)
+        ioc_infos = []
+        while more:
+
+            ret_val, response = self._make_rest_call_helper_oauth2(action_result, CROWDSTRIKE_GET_CUSTOM_INDICATORS_ENDPOINT, params=api_data)
 
-        ret_val, resp_json = self._make_rest_call_helper_oauth2(action_result, CROWDSTRIKE_GET_CUSTOM_INDICATORS_ENDPOINT, params=params, method='get')
+            if (phantom.is_fail(ret_val)):
+                return action_result.get_status()
 
-        if (phantom.is_fail(ret_val)):
-            return action_result.get_status()
+            ioc_infos.extend(response["resources"])
 
-        action_result.add_data(resp_json)
+            offset = response["meta"]["pagination"]["offset"]
+            total = response["meta"]["pagination"]["total"]
 
-        summary = action_result.update_summary({})
-        summary['total_host_groups'] = action_result.get_data_size()
+            if (total):
```

Comment:
please remove the parentheses around the overall conditional for this and all the other `if` statements

Reply:
Done